### DISCUSSION TITLE
Allow projects using the CMake integration to automatically select an appropriate 'static' triplet

### DISCUSSION THREAD

**Discussion by fredemmott** [microsoft/vcpkg#42909] [Ideas]
With MSBuild I can do something like this:

```xml
 <PropertyGroup Label="Globals">
 <!-- .... -->
 <VcpkgTriplet Condition="'$(Platform)'=='Win32'">x86-windows-static</VcpkgTriplet>
 <VcpkgTriplet Condition="'$(Platform)'=='x64'">x64-windows-static</VcpkgTriplet>
</PropertyGroup>
```

it would be nice if I could do something similar when using the CMake project, instead of requiring the user to specify `-DVCPKG_TARGET_TRIPLET`

This isn't current possible as:
- CMake doesn't make the architecture available until *after* `project()`
- When using the vcpkg toolchain file, the triplet must be set *before* `project()`

**Comment by FrankXie05** [comment]
You can set whether to enable the `static` option in the project properties of `msbuild`, and you can manually enter to select the existing installation triples.

![Image](https://github.com/user-attachments/assets/40ae8579-141e-45dc-8b83-ccc5c8757b64)

**Comment by fredemmott** [comment]
@FrankXie05 this post is about cmake, not msbuild, and it is a vcpkg feature request, not a general usage question - i used vcpkg’s msbuild support as an example. your response is answering a different question to what I asked.

**Comment by FrankXie05** [comment]
If you choose the vcpkg build triplet based on the project, it is indeed more convenient.

However, I think this is almost impossible to achieve. Without calling vcpkg in the cmake project, `toolchain` is the best integration method.
If you want to adapt, you must call `vcpkg` in the cmake project.

Just like the `project()` mechanism you mentioned, we can't bypass it.

**Comment by FrankXie05** [comment]
I wish I could find a way to do this. But it seems to me that this is not possible, so I'm turning this into a discussion.
